addons/web/static/src/core/l10n/dates.js

```diff
@@ -278,7 +278,18 @@ export function parseDate(value, options = {}) {
     if (!value) {
         return false;
     }
-    return parseDateTime(value, options).startOf("day");
+
+    try {
+        return parseDateTime(value, options).startOf("day");
+    } catch (error) {
+        if (!options.format) {
+            return parseDateTime(value, {
+                ...options,
+                format: localization.dateFormat,
+            }).startOf("day");
+        }
+        throw error;
+    }
```

Comment:
While this piece of code works fine I find it a bit overkill.

I want a more minimalistic approach, please apply the following suggestion (and of course feel free to discuss if you do not agree):
```suggestion
    return parseDateTime(value, { format: localization.dateFormat, ...options}).startOf("day");
```

Reply:
Sadly I tried this here https://github.com/odoo-dev/odoo/commit/6c0381d4df9c303dbdeab9a4b32bb69fdce6b8cb and some tests in `dates utils compatibility with legacy > parseDate` are not passing

```
QUnit test failed: utils > dates utils compatibility with legacy > parseDate :
	message: ""10101010101010" is parsed as expected. [new]"
	expected: "1010-10-10T00:00:00.000Z"
	actual: "undefined"
QUnit test failed: utils > dates utils compatibility with legacy > parseDate :
	message: ""3101198508" is parsed as expected. [new]"
	expected: "1985-01-31T00:00:00.000Z"
	actual: "undefined"
QUnit test failed: utils > dates utils compatibility with legacy > parseDate :
	message: ""310119850833" is parsed as expected. [new]"
	expected: "1985-01-31T00:00:00.000Z"
	actual: "undefined"
QUnit test failed: utils > dates utils compatibility with legacy > parseDate :
	message: ""31/01/1985 08" is parsed as expected. [new]"
	expected: "1985-01-31T00:00:00.000Z"
	actual: "undefined"
```

Reply:
For the test `Search > SearchBar > search date and datetime fields. Support of timezones`
You can remove this line: https://github.com/odoo/odoo/blob/14eddf9fde5480d283245f4f3506bbe1aa3fa52d/addons/web/static/tests/search/search_bar_tests.js#L250
The "birthday" choice disappears from the SearchBar component because the input value is now no more parsable through parseDate, so choices in the dropdown differs from originally. IMHO this is better.

For the failing assertions in `dates utils compatibility with legacy > parseDate`, you can change the expected values and we'll discuss this before merging.
Here, IMHO again, the test cases are a bit convoluted. I don't think a user would write down a time part in a date field.

cc @aab-odoo what do you think ? There may be one use case we could discuss: if a user paste a value from its clipboard. But I hardly see why he would copy the time part. And if he does it i.e. from some kind of spreadsheet program, they generally offer multiple ways to format their data easily.